### PR TITLE
Update configUtil and its usages to get configuration values from core APIs in salesforcedx-sobjects-faux-generator

### PR DESCRIPTION
**Pull request by klewis-sfdc**
### What does this PR do?
Updates the salesforcedx-sobjects-faux-generator package to use core APIs when retrieving configuration values.

### What issues does this PR fix or reference?
@W-11599361@

### Functionality Before
Default username was retrieved by directly accessing the .sfdx directory.

### Functionality After
Default username value is retrieved using ConfigAggregator from salesforce/core lib.

### CODE REVIEW

File: packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts [hunk 1]
```diff
@@ -9,23 +9,33 @@ import {
   ConfigAggregator,
   ConfigFile,
   ConfigValue,
-  StateAggregator
+  OrgConfigProperties
 } from '@salesforce/core';
 import * as path from 'path';
 
-const defaultUserNameKey = 'defaultusername';
+async function getConfigAggregator(
```
Comment (gbockus-sf) [inline packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts:16]: Nit: can we add a TODO with maybe the consolidation ticket number to ensure we find all these when we do the consolidation ticket?

File: packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts [hunk 2]
```diff
@@ -9,23 +9,33 @@ import {
...
+  // so that ConfigAggregator reads the local project values
+  process.chdir(projectPath);
+  const configAggregator = await ConfigAggregator.create();
+  // Change the current working directory back to what it was
+  // before returning
+  process.chdir(origCurrentWorkingDirectory);
+  return configAggregator;
+}
 
 export class ConfigUtil {
```
Comment (gbockus-sf) [inline packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts:30]: sorry dumb q.  Can we just update this package to remove this configUtil and use the one in utils? Or is that the scope creep we're hoping to avoid?

File: packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts [hunk 3]
```diff
@@ -9,23 +9,33 @@ import {
...
-      defaultUserNameKey
-    )) as string;
-    const info = await StateAggregator.getInstance();
-    const username = info.aliases.resolveValue(defaultUserName);
-    return username;
+    const configAggregator = await getConfigAggregator(projectPath);
+    const defaultUsernameOrAlias = configAggregator.getPropertyValue(
+      OrgConfigProperties.TARGET_ORG
+    );
+    return (defaultUsernameOrAlias as string) || undefined;
```
Comment (gbockus-sf) [inline packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts:38]: same nit as the other PR where `return defaultUsernameOrAlias ? String(defaultUsernameOrAlias) : undefined` is preferred.

File: packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts [hunk 4]
```diff
@@ -9,23 +9,33 @@ import {
...
   }
```
Comment (gbockus-sf) [inline packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts:39]: Is there no associated test for this function?

File: packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts (hunk 3 above)
Comment (klewis-sfdc) [reply packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts:38]: OK, I'll update all of these!

File: packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts (hunk 2 above)
Comment (klewis-sfdc) [reply packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts:30]: Beginning to do that sort of work is where I started running into issues before.  However, this is a smaller scoped set of changes than last time, and the build has evolved.  I'll try it and report back.

Comment (klewis-sfdc) [reply packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts:30]: This package doesn't have a dep on the utils package like the core package does.  Could that be why importing from utils wouldve caused issues last time?

Comment (klewis-sfdc) [reply packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts:30]: https://salesforce-internal.slack.com/archives/G02JDKH7W9K/p1660866255590169

File: packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts (hunk 4 above)
Comment (klewis-sfdc) [reply packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts:39]: The username is used to get the instance of Connection, and it looks like the connection is stubbed in the tests:

**Review by gbockus-sf — APPROVED** [review]
This looks good to go.  Lemme QE real quick that I can still refresh object and it'll be :shipit: time

**Review by gbockus-sf — APPROVED** [review]
Passed by QE with the updates: 
✅ Can refresh sobjects

There is a known issue if you switch users the configAggregator is caching the current username.  Ken is addressing that separately.

File: packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts [hunk 5]
```diff
@@ -42,6 +42,10 @@ export class ConfigUtil {
     // so that ConfigAggregator reads the local project values
     process.chdir(projectPath);
     const configAggregator = await ConfigAggregator.create();
+    // ConfigAggregator caches instances internally by working directory
+    // path.  Clling reload ensures that this instance of ConfigAggregator
+    // has the latest values from the config file.
+    configAggregator.reload();
```
Comment (klewis-sfdc) [inline packages/salesforcedx-sobjects-faux-generator/src/generator/configUtil.ts:48]: See vid comment: https://salesforce-internal.slack.com/archives/G02JDKH7W9K/p1661361086759919

**Review by gbockus-sf — APPROVED** [review]
:shipit:
